cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerDeployClientConfigProvider.java

```diff
@@ -258,4 +253,17 @@ private void enhanceRequestParams(Map<String, List<String>> headers, List<Pair>
     ExecutorService createExecutor() {
         return Executors.newSingleThreadExecutor();
     }
+
+    @VisibleForTesting
+    void addPathSegmentsFromDefaultUrl(
+            HttpUrl defaultHttpUrl, HttpUrl.Builder httpUrlBuilder) {
+        List<String> defaultPathSegments = defaultHttpUrl.pathSegments();
+        int apiIndex = defaultPathSegments.indexOf("api");
```

Comment:
it might be an edge case or never should happen but you should handle if this returns `-1`, so it's not found. just in case. 
also some log would be nice here to see what's going on, what we modifying and what's the result